sysvabi64/sysvabi64.rst

```diff
@@ -749,12 +749,27 @@ open-source compilers.
 Medium code model
 -----------------
 
-Symbols smaller than 64KiB (value TBD) are treated like the small code model.
-Symbols which are larger than 64KiB, are ``common`` or of unknown size use
-sections ``.ldata``, ``.lbss`` and ``.lrodata``. These sections are placed after
-all other sections and can be unlimited in size. Symbols in them are addressed
-via the GOT. Linkers may relax GOT indirections into PC-relative addressing for
-non pre-emptible symbols within 4GiB.
+
+The medium code model is suitable for programs that do not have large amounts
+of executable code, but do contain large static data objects. The medium code
+model separates data into small data and large data. Small data is addressed
+in the same way as the small code model, with the same maximum size
+restrictions as the small code model. Large data is addressed via the GOT and
+does not have a maximum size limit.
+
+Data is defined as large data if any of the following are true:
+
+- The data is >= 64 KiB in size.
+- The data is of unknown size.
+- The data has type ``SHN_COMMON``.
```

Comment:
Being picky the symbol type is `STT_COMMON`` but the ELF spec refers to ``SHN_COMMON`` Symbols (``STT_COMMON`` symbols in objects have ``st_index`` = ``SHN_COMMON``. I take your point that people not familiar with ELF might not know what that means. I've made a suggestion below:

The data will be represented by an unallocated common block, described by a ``SHN_COMMON`` symbol.

Reply:
That's clearer indeed.